designs/2024-baseline-support/README.md

```diff
@@ -0,0 +1,381 @@
+
+``` bash
+eslint --suppress-all ./src
+```
+
+### Suppressing violations of a specific rule
+
+A new option `--suppress-rule [RULE1]` will be introduced to ESLint CLI. When provided, the existing suppressions file will be updated to include any existing violation of the provided rule. The suppressions file will be created if not already exists. Note that this is option can accept an array of string values.
+
+``` bash
+eslint --suppress-rule '@typescript-eslint/no-explicit-any' --suppress-rule '@typescript-eslint/member-ordering' ./src
+```
+
+### Changing the location of the suppressions file
+
+A new option `--suppressions-location` will be introduced to ESLint CLI. When provided, the suppressions file will be loaded and saved to the provided location. Note that this is a string flag option (value is required).
+
+``` bash
+eslint --suppress-all --suppressions-location /home/user/project/mycache ./src
+```
+
+### Maintaining a lean suppressions file
+
+When working with suppressed violations, it's possible to address a violation without updating the suppressions file. This oversight can allow new violations to go unnoticed. To prevent this, eslint can exit with an error code if there are outdated (unmatched) suppressions.
+
+Consider the following scenario:
+
+* The developer runs `eslint --supress-all ./src` to create the suppressions file.
+* Running `eslint ./src` reports no violations and exits with status 0.
+* After fixing a violation, the suppressions file still contains the now-resolved violation.
+* Running `eslint ./src` again reports no violations but exits with a non-zero status code, indicating the suppressions file needs updating.
+
+To address this, a new option `--prune-suppressions` will be introduced to ESLint. This boolean flag removes resolved violations from the suppressions file without adding new ones, unlike `--suppress-all`.
+
+``` bash
+eslint --prune-suppressions ./src
+eslint --prune-suppressions --suppressions-location /home/user/project/mycache ./src
+```
+
+### Execution details
+
+The suggested solution always compares against the existing suppressions file, typically `.eslint-suppressions.json`, unless `--suppressions-location` is specified.  This makes it easier for existing and new projects to adopt this feature without the need to adjust scripts in `package.json` and CI/CD workflows. 
+
+To perform the comparison, we will go through each result and message from `ESLint.lintFiles`, checking each error `(severity == 2)` against the suppressions file. By design, we ignore warnings since they don't cause eslint to exit with an error code and serve a different purpose. If the file and rule are listed in the suppressions file, we can move the message to `LintResult#suppressedMessages` and ignore the result message.
```

Comment:
[`SuppressedLintMessage`](https://eslint.org/docs/latest/integrate/nodejs-api#-suppressedlintmessage-type) should have an additional property `suppressions`. We should specify what value it will have in this case. I think it can be:

```js
suppressions: [{ kind: "file", justification: "" }]
```

Reply:
@mdjermanovic can you please clarify what would be the purpose of this new property?

Reply:
It describes the reason why the lint message is suppressed. It's a mandatory property in `SuppressedLintMessage` type (objects in `LintResult#suppressedMessages`). Without it, integrations and formatters that use `LintResult#suppressedMessages` could break.

Reply:
Thanks @mdjermanovic . I was under the impression that you were referring to a new property for some reason. That is clear now.